vllm/model_executor/sampling_metadata.py

```diff
@@ -249,16 +249,20 @@ def _prepare_seq_groups(
     for i, seq_group_metadata in enumerate(seq_group_metadata_list):
         seq_ids = seq_group_metadata.seq_data.keys()
 
+        sampling_params = seq_group_metadata.sampling_params
         if cache is not None:
             sample_obj = cache.get_cached_seq_group_to_sample(len(seq_ids))
 
+            if sample_obj.sampling_params is not None:
+                sampling_params.logits_processors = \
+                    sample_obj.sampling_params.logits_processors
+
```

Comment:
Does the GPU backend have the problem? In general, the ```cache``` is only used for reducing Python object allocation overhead, the content of the cache should be overwritten rather than reusing. Maybe the ```logits_processors``` should be filled at some other place.

Reply:
Thanks, I realized that this Bug only occurs when invoking async_execute_model, the main reason is that the sampling_param gets copied when submitting to mp.Queue.

I agree that putting under SamplingMetaCache is not a good way, still trying to find a better way to fix this error.